libmount/src/context_mount.c

```diff
@@ -334,10 +334,14 @@ static int fix_optstr(struct libmnt_context *cxt)
 						name,
 						val ? val + valsz :
 						      name + namesz);
-			} else if (se_fix && val && valsz)
-				/* translate selinux contexts */
-				rc = mnt_optstr_fix_secontext(&fs->fs_optstr,
-							val, valsz, &next);
+			} else if (se_fix && val && valsz) {
+				if (strncmp(name, "rootcontext", strlen("rootcontext")) == 0 && strcmp(val, "!auto") == 0)
```

Comment:
It would be better to use `sizeof("rootcontext") - 1` to keep it in the preprocessor rather than use a function strlen().

You also need strncmp() for the value (see valsz), as the value does not have to be zero-terminated.